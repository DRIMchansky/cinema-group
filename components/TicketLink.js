// packages
import styled from 'styled-components'
// components
import Button from '@material-ui/core/Button'
import ConfirmationNumberIcon from '@material-ui/icons/ConfirmationNumber'
// utils
import { colors } from '../utils/cssConfig'

const TicketLink = () => {
  return (
    <StyledWrapper>
      <StyledTicketLink disableElevation disableRipple href="/">
        <StyledLabel>в кино</StyledLabel>
        <ConfirmationNumberIcon />
      </StyledTicketLink>
    </StyledWrapper>
  )
}

const StyledWrapper = styled.div`
  position: absolute;
  right: 0;
  top: 50%;
  transform: translateY(-50%);
`

const StyledTicketLink = styled(Button)`
  color: ${colors.white} !important;
`

const StyledLabel = styled.span`
  margin-right: 5px;

  @media (max-width: 600px) {
    display: none;
    visibility: hidden;
  }
`

export default TicketLink
