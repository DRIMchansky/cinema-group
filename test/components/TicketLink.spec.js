import React from 'react'
import renderer from 'react-test-renderer'
import TicketLink from '../../components/TicketLink'

test('TicketLink renders correctly', () => {
  const tree = renderer.create(<TicketLink />).toJSON()

  expect(tree).toMatchSnapshot()
})
