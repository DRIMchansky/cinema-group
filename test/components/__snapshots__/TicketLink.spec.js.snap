// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketLink renders correctly 1`] = `
.c0 {
  position: absolute;
  right: 1rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c1 {
  color: #fff !important;
  background: #324082 !important;
}

.c2 {
  margin-right: 5px;
}

@media (max-width:600px) {
  .c2 {
    display: none;
    visibility: hidden;
  }
}

<div
  className="c0"
>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiButton-root MuiButton-text c1 MuiButton-disableElevation"
    href="/"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiButton-label"
    >
      <span
        className="c2"
      >
        в кино
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 10V6c0-1.11-.9-2-2-2H4c-1.1 0-1.99.89-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2s.9-2 2-2zm-9 7.5h-2v-2h2v2zm0-4.5h-2v-2h2v2zm0-4.5h-2v-2h2v2z"
        />
      </svg>
    </span>
  </a>
</div>
`;
